### ISSUE TITLE
Remove analyzer suppressions from libraries in repo

### ISSUE THREAD

**Issue by annelo-msft** [Azure/azure-sdk-for-net#38268]
A number of our analyzer warnings are suppressed in various libraries.  This tracks the work to evaluate those suppressions and remove them where not needed, including updating the analyzers themselves to be more effectively scoped.

- [ ] [AZC0002](https://grep.app/search?q=azc0002&filter[repo][0]=Azure/azure-sdk-for-net)
- [ ] [AZC0003](https://grep.app/search?q=azc0003&filter[repo][0]=Azure/azure-sdk-for-net)

**Comment by pallavit** [comment]
Do you have list of SDKs that need to be updated. I think this is something we should definitely push towards for our SDKs this MQ.

**Comment by annelo-msft** [comment]
> Do you have list of SDKs that need to be updated?

Yep!  Each of the links above should show all the search hits for suppressions in our repo - and each suppression is embedded in a library we should address.  Let me know if you'd like more information than this - it would be a simple exercise to types this up manually, but the search results will always be the most up-to-date.

**Comment by pallavit** [comment]
Ahh my bad, I missed that. This is good, thanks!
